tracer/jaeger_wavefront.go

```diff
@@ -0,0 +1,187 @@
+package tracer
+
+import (
+	"bytes"
+	"errors"
+	"github.com/google/uuid"
+	"github.com/opentracing/opentracing-go"
+	"strconv"
+	"strings"
+)
+
+const (
+	BAGGAGE_PREFIX        = "baggage-"
+	TRACE_ID_KEY          = "trace-id"
+	PARENT_ID_KEY         = "parent-id"
+	SAMPLING_DECISION_KEY = "sampling-decision"
+)
+
+type JaegerWavefrontPropagator struct {
+	traceIdHeader string
+	baggagePrefix string
+	tracer        *WavefrontTracer
+}
+
+type Setter func(*JaegerWavefrontPropagator)
+
+func (j *JaegerWavefrontPropagator) WithBaggagePrefix(baggagePrefix string) {
```

Comment:
Convert both these `With` functions to `JaegerOption`'s and let's have a single constructor . `NewJaegerWavefrontPropagator` that takes optional `JaegerOption`'s and invokes them over a for loop if they are set.

So